packages/repository/src/repositories/legacy-juggler-bridge.ts

```diff
@@ -461,3 +464,23 @@ export class DefaultCrudRepository<
     return models.map(m => this.toEntity<R>(m));
   }
 }
+
+/**
+ * Default implementation of CRUD repository using legacy juggler model
+ * and data source with beginTransaction() method for connectors which
+ * support Transactions
+ */
+
+export class DefaultTransactionalRepository<
+  T extends Entity,
+  ID,
+  Relations extends object = {}
+> extends DefaultCrudRepository<T, ID, Relations>
+  implements TransactionalRepository<T, ID, Relations> {
+  async beginTransaction(
+    options?: IsolationLevel | Options,
+  ): Promise<Transaction> {
+    const dsOptions: juggler.IsolationLevel | Options = options || {};
+    return (await this.dataSource.beginTransaction(dsOptions)) as Transaction;
```

Comment:
I find this suspicious, why is the type cast from `juggler.Transaction` to `Transaction` necessary? I expected those interfaces to be the same, what am I missing?

Reply:
Yea that was my expectation as well, but the `juggler.Transaction` still have the Promise/Callback variants of the Transaction methods (e.g. https://github.com/strongloop/loopback-datasource-juggler/blob/master/types/transaction-mixin.d.ts#L43-L47).

Reply:
Makes sense. Can you please add a code comment to explain why the cast is needed?

BTW I think you posted a link pointing to a wrong place, did you mean this insted?

https://github.com/strongloop/loopback-datasource-juggler/blob/ad4edc6a1569eace4876c2c3e5b32e4661d4d6eb/types/transaction-mixin.d.ts#L11-L22

Reply:
Yeah I meant to post that sorry. Will do 👍

Reply:
Should it be the following now?
```ts
return this.dataSource.beginTransaction(dsOptions) as Promise<Transaction>;
```